### PR TITLE
Update to GridTools C++ v2.2.0

### PR DESCRIPTION
**Pull request by havogt**
Updates code generator to changes from GridTools v2.1.0 to GridTools v2.2.0.
- Require c++17 (consequence: drop support for CUDA < 11)
- Update to new `make` pattern.

### CODE REVIEW

**Comment by havogt** [comment]
bors try

**Comment by havogt** [comment]
bors try-

**Comment by havogt** [comment]
bors try

**Comment by havogt** [comment]
bors try

**Comment by havogt** [comment]
bors try

**Comment by havogt** [comment]
@jdahm Are you ready everywhere to use c++17, especially to drop support for CUDA < 11?

**Comment by havogt** [comment]
bors try

**Comment by havogt** [comment]
bors try

**Comment by havogt** [comment]
bors try

**Review by jdahm — APPROVED** [review]
Thanks for updating it! This should work for Pace.

File: src/gt4py/backend/dace_backend.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ class DaCeComputationCodegen:
                 const int __J = domain[1];
                 const int __K = domain[2];
                 ${name}_t dace_handle;
-                auto allocator = gt::sid::make_cached_allocator(&${allocator}<char[]>);
+                auto allocator = gt::sid::cached_allocator(&${allocator}<char[]>);
```
Comment (gronerl) [inline src/gt4py/backend/dace_backend.py:186]: Is this just updated API or a semantic change?

File: src/gt4py/gt_src_manager.py [hunk 2]
```diff
@@ -22,8 +22,15 @@
...
-_GRIDTOOLS_GIT_BRANCHES = {1: "v1.1.4", 2: "v2.1.0"}
+
+_GRIDTOOLS_GIT_REPO = {
+    1: "https://github.com/GridTools/gridtools.git",
+    2: "https://github.com/GridTools/gridtools.git",
+}  # TODO reset to main repo
+_GRIDTOOLS_GIT_BRANCHES = {
+    1: "v1.1.4",
+    2: "v2.2.0",
+}
```
Comment (gronerl) [inline src/gt4py/gt_src_manager.py:33]: There's nothing in gt4py (capable of) using GT 1 anymore, might as well remove the option from the source manager.  Not necessarily the scope of this PR though.

**Review by gronerl — APPROVED** [review]
:+1:

File: src/gt4py/gt_src_manager.py (hunk 2 above)
Comment (havogt) [reply src/gt4py/gt_src_manager.py:33]: Yes, and thanks to pointing to this line which contains a TODO that I should remove...

File: src/gt4py/backend/dace_backend.py (hunk 1 above)
Comment (havogt) [reply src/gt4py/backend/dace_backend.py:186]: updated API

File: src/gt4py/gt_src_manager.py [hunk 3]
```diff
@@ -22,8 +22,15 @@
 
 
 _DEFAULT_GRIDTOOLS_VERSION = 2
-# TODO: GT2 release with CUDA SID adapter
-_GRIDTOOLS_GIT_BRANCHES = {1: "v1.1.4", 2: "v2.1.0"}
+
+_GRIDTOOLS_GIT_REPO = {
+    1: "https://github.com/GridTools/gridtools.git",
+    2: "https://github.com/GridTools/gridtools.git",
+}  # TODO reset to main repo
```
Comment (havogt) [inline src/gt4py/gt_src_manager.py:29]:
```suggestion
}
```

File: src/gt4py/gt_src_manager.py (hunk 2 above)
Comment (egparedes) [reply src/gt4py/gt_src_manager.py]: I agree with @gronerl that we should remove GT1 handling in source manager. If it's not done in this PR then I think it's worth to add a TODO at least.

**Pull request merged**